src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -107,6 +107,17 @@ private bool EmitNupkg(PublishRoot root)
                 root.Operations.Delete(TargetPath);
             }
 
+            // If this is a wrapper project, we need to generate a lock file before building it
+            if (IsWrappingAssembly())
+            {
+                var success = Restore(root, publishProject: this, restoreDirectory: project.ProjectDirectory)
```

Comment:
If you never restore your wrapper project, then obviously you cannot build the wrapper project and `dnu publish` will fail.

I kind of understand your concern because previously, when you publish an app without any package installed, the publish can still be successful with warnings say that some package dependencies cannot be found. However, if we check this in, you cannot publish without restoring packages first (when you have a wrapper project reference).